<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Aitor Gonzalez" />
  <title>Tagoos Documentation</title>
  <link rel="stylesheet" href="buttondown.css" type="text/css" />
</head>
<body>
<div id="header">
<h1 class="title">Tagoos Documentation</h1>
<h2 class="author">Aitor Gonzalez</h2>
</div>
<div id="TOC">
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#data-preprocessing">Data preprocessing</a></li>
<li><a href="#model-learning-and-dbsnp-variant-prediction">Model learning and DBSNP variant prediction</a><ul>
<li><a href="#grasp-database">GRASP database</a></li>
</ul></li>
<li><a href="#other-phenotypes">Other phenotypes</a><ul>
<li><a href="#height">Height</a></li>
<li><a href="#blood-pressure">Blood pressure</a></li>
<li><a href="#serum-urate">Serum Urate</a></li>
<li><a href="#body-mass-index">Body-mass-index</a></li>
<li><a href="#transmission-distortion">Transmission distortion</a></li>
</ul></li>
</ul>
</div>
<h1 id="overview"><a href="#TOC">Overview</a></h1>
<p>This is an overview of the tagoos workflow. There are two large parts.</p>
<ol style="list-style-type: decimal">
<li>Process 1000 genome and dbsnp variants with NGS data:</li>
<li>Create a model and score the dbsnp variants</li>
</ol>
<p>In order to create a model, these parameters are needed:</p>
<ul>
<li>A positive RSID list</li>
<li>(Optional) A list of negative random SNPs from the 1000 genome database</li>
<li>(Optional) An annotation file in bed format: annotationcorr</li>
<li>(Optional) A precomputed index file: index, index2 or index3</li>
</ul>
<p>In order to carry out there are two large parts</p>
<ol style="list-style-type: decimal">
<li>The preprocessing step where 1000 genome and dbsnp variants are processed and annotated with NGS data</li>
<li>The modeling and scoring of the dbsnp variants.</li>
</ol>
<p>The first part of the workflow is carried out in the following manner</p>
<ul>
<li>split annotation: <strong>split_annotation.yml</strong> :ref:<code>genindex</code></li>
<li>annotate 1000 genome data: <strong>preannotate.yml</strong> with 1000 genome parameters</li>
<li>annotate dbsnp: <strong>preannotate.yml</strong> with dbsnp parameters</li>
<li>LD and index calculation of 1000 genome data: <strong>genome1k.yml</strong></li>
</ul>
<p>The second part of the workflow is carried out in the following manner</p>
<ul>
<li>Create a specific model: <strong>Snakemake.yml</strong></li>
<li>Score the dbsnp database with a specific model: <strong>score_dbsnp.yml</strong></li>
</ul>
<h1 id="data-preprocessing"><a href="#TOC">Data preprocessing</a></h1>
<ul>
<li>Split the annotation in different chromosomes using the <strong>split_annotation.yml</strong> snakefile</li>
</ul>
<pre><code>export ANNOT_LABEL=annotationcorr
export ANNOT_1COL_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}/${ANNOT_LABEL}_1col.bed
export ANNOT_1COL_DIR=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}
#
export CHROM=$(seq 1 22)
export NBCHROM=`python -c &quot;import os; print(len(os.getenv(&#39;CHROM&#39;).split()))&quot;`
time snakemake -s snakefile/split_annotation.yml -p -j $NBCHROM -c &quot;qsub -X -V -q tagc -l nodes=1:ppn={threads} -e $ANNOT_1COL_DIR/stderr.log -o $ANNOT_1COL_DIR/stdout.log&quot; -d $ANNOT_1COL_DIR -pn</code></pre>
<ul>
<li>Intersect/annotate the 1000 genome variants using the <strong>preannotate.yml</strong> snakefile</li>
</ul>
<pre><code>export ANNOT_LABEL=annotationcorr
export ANNOT_1COL_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}/${ANNOT_LABEL}_1col.bed
#
export CHROM=$(seq 1 22)
export SNP_DIR=$HOME/data/2015_svmgwas/data/variant/1000genomes/eur
export SCRIPTDIR=$HOME/data/2015_svmgwas/repositories/tagoos/script
export NBCHROM=`python -c &quot;import os; print(len(os.getenv(&#39;CHROM&#39;).split()))&quot;`
time snakemake -s snakefile/preannotate.yml -j $NBCHROM -c &quot;qsub -X -V -q tagc -l nodes=1:ppn={threads} -e tmp/stderr.log -o tmp/stdout.log&quot; -d $SNP_DIR annotated_tsv -pn</code></pre>
<ul>
<li>Compute the correlated and index variants of 1000 genome data with the <strong>genome1k.yml</strong> snakefile</li>
</ul>
<pre><code>export CHROM=$(seq 21 22)
export GENOME1K_DIR=$HOME/data/2015_svmgwas/data/variant/1000genomes/eur
export SCRIPTDIR=$HOME/data/2015_svmgwas/repositories/tagoos/script
export NBCHROM=`python -c &quot;import os; print(len(os.getenv(&#39;CHROM&#39;).split()))&quot;`
time snakemake -s genome1k.yml -p -j $NBCHROM -c &quot;qsub -X -V -q tagc -l nodes=1:ppn={threads} -e tmp/stderr.log -o tmp/stdout.log&quot; -d $GENOME1K_DIR -pn</code></pre>
<ul>
<li>Download the dbsnp variants using the <strong>download_dbsnp.yml</strong> snakefile</li>
</ul>
<pre><code>export CHROM=$(seq 1 22)
export DBSNP_DIR=$HOME/data/2015_svmgwas/data/variant/dbsnp
export NBCHROM=`python -c &quot;import os; print(len(os.getenv(&#39;CHROM&#39;).split()))&quot;`
time snakemake -s snakefile/download_dbsnp.yml -j $NBCHROM -c &quot;qsub -X -V -d $PWD -q tagc -l nodes=1:ppn=
{threads} -e stderr.log -o stdout.log&quot; -d $PWD -pn</code></pre>
<ul>
<li>Intersect/annotate dbsnp variants using the <strong>preannotate.yml</strong> snakefile like for the 1000 genome variants</li>
</ul>
<pre><code>export ANNOT_LABEL=annotationcorr
export ANNOT_1COL_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}/${ANNOT_LABEL}_1col.bed
#
export CHROM=$(seq 1 22)
export SNP_DIR=$HOME/data/2015_svmgwas/data/variant/dbsnp
export SCRIPTDIR=$HOME/data/2015_svmgwas/repositories/tagoos/script
export NBCHROM=`python -c &quot;import os; print(len(os.getenv(&#39;CHROM&#39;).split()))&quot;`
time snakemake -s snakefile/preannotate.yml -p -j $NBCHROM -c &quot;qsub -X -V -q tagc -l nodes=1:ppn={threads} -e $SNP_DIR/stderr.log -o $SNP_DIR/stdout.log&quot; -d $SNP_DIR -pn</code></pre>
<h1 id="model-learning-and-dbsnp-variant-prediction"><a href="#TOC">Model learning and DBSNP variant prediction</a></h1>
<h2 id="grasp-database"><a href="#TOC">GRASP database</a></h2>
<p>In this example we create a model for</p>
<ul>
<li>Associated GRASP variants (p-val &lt;10‚Å∏)</li>
<li>100000 random variants for the 1000 genome DB</li>
<li>The <strong>annotationcorr</strong> variants</li>
<li>The <strong>index2</strong> variants</li>
</ul>
<p>The important paramenters are set before the hash character</p>
<pre><code>export TAG_RSID_POS=$HOME/data/2015_svmgwas/data/variant/GRASP/GRASP108.rsid
export POS_LABEL=GRASP108
export NEG_LABEL=1kg100000
export ANNOT_LABEL=annotationcorr
export INDEX_LABEL=index2
export ANNOT_1COL_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}/${ANNOT_LABEL}_1col.bed
#
export OUTDIR=$PWD/out/${POS_LABEL}/${NEG_LABEL}_${ANNOT_LABEL}_${INDEX_LABEL}
export POSDIR=$PWD/out/${POS_LABEL}/pos/${ANNOT_LABEL}_${INDEX_LABEL}
export NEGDIR=$PWD/out/neg/${NEG_LABEL}_${ANNOT_LABEL}_${INDEX_LABEL}
export TAG_RSID_NEG=$HOME/data/2015_svmgwas/data/variant/1000genomes/${NEG_LABEL}.rsid
export CHROM=$(seq 1 22)
export GENOME1K_DIR=$HOME/data/2015_svmgwas/data/variant/1000genomes/eur
export ANNOTATION_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}_1col.bed 
export SCRIPTDIR=$HOME/data/2015_svmgwas/repositories/tagoos/script
export NBCHROM=`python -c &quot;import os; print(len(os.getenv(&#39;CHROM&#39;).split()))&quot;`
time snakemake -s ${TAGOOS}/snakefile/Snakemake.yml -p -j $NBCHROM -c &quot;qsub -X -V -d $OUTDIR -q tagc -l nodes=1:ppn={threads} -e $OUTDIR/stderr.log -o $OUTDIR/stdout.log&quot; -d $OUTDIR -pn</code></pre>
<p>After creating the model, we can score all the DBSNP variants using the following commands. Again the important parameters are set before the hash character</p>
<pre><code>export RSID_LABEL=dbsnp
export POS_LABEL=GRASP108
export NEG_LABEL=1kg100000
export ANNOT_LABEL=annotationcorr
export INDEX_LABEL=index2
export ANNOT_1COL_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}/${ANNOT_LABEL}_1col.bed
#
export ANNOTATION_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}_1col.bed 
export DBSNP_DIR=$HOME/data/2015_svmgwas/data/variant/dbsnp
export MODEL_PKL=$PWD/out/${POS_LABEL}/${NEG_LABEL}_${ANNOT_LABEL}_${INDEX_LABEL}/model.pkl
export VARIABLE=$PWD/out/${POS_LABEL}/${NEG_LABEL}_${ANNOT_LABEL}_${INDEX_LABEL}/variable.txt
export SCRIPTDIR=$HOME/data/2015_svmgwas/repositories/tagoos/script
export OUTDIR=$PWD/out/${POS_LABEL}/score_${RSID_LABEL}_${POS_LABEL}_${NEG_LABEL}_${ANNOT_LABEL}_${INDEX_LABEL}
#export CHROM=$($HOME/data/2015_svmgwas/data/hcomp/get_record  $HOME/data/2015_svmgwas/data/variant/1000genomes/eur/eur.peak.bed.idx $HOME/data/2015_svmgwas/data/variant/1000genomes/eur/eur.peak.bed -f $RSID_PATH |cut -f1 |tr -d &quot;chr&quot; |sort -u -k1n)
export CHROM=$(seq 1 22)
export NBCHROM=`python -c &quot;import os; print(len(os.getenv(&#39;CHROM&#39;).split()))&quot;`
time snakemake -s ${TAGOOS}/snakefile/score_dbsnp.yml -j ${NBCHROM} -c &quot;qsub -X -V -d $OUTDIR -q tagc -l nodes=1:ppn={threads} -e $OUTDIR/stderr.log -o $OUTDIR/stdout.log&quot; -d $OUTDIR -pn</code></pre>
<h1 id="other-phenotypes"><a href="#TOC">Other phenotypes</a></h1>
<p>We have shown the detailed parameters for the GRASP variants. A few other phenotypes are working well with the following parameters.</p>
<h2 id="height"><a href="#TOC">Height</a></h2>
<pre><code>export TAG_RSID_POS=/cobelix/gonzalez/data/2015_svmgwas/data/variant/literature_association/traits/heightMergedNature2010NatGen2014.rsid
export POS_LABEL=heightMergedNature2010NatGen2014
export NEG_LABEL=1kg100000
export ANNOT_LABEL=annotationcorr
export INDEX_LABEL=index3
export ANNOT_1COL_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}/${ANNOT_LABEL}_1col.bed</code></pre>
<h2 id="blood-pressure"><a href="#TOC">Blood pressure</a></h2>
<pre><code>export TAG_RSID_POS=$HOME/data/2015_svmgwas/data/variant/literature_association/traits/bloodPressureMerged.rsid
export POS_LABEL=bloodPressureMerged
export NEG_LABEL=1kg100000
export ANNOT_LABEL=annotationcorr
export INDEX_LABEL=index3
export ANNOT_1COL_BED=$HOME/data/2015_svmgwas/data/annotation_ngs_based/${ANNOT_LABEL}/${ANNOT_LABEL}_1col.bed</code></pre>
<p>On the other hand, these phenotypes are not working well but can be kept as negative control</p>
<h2 id="serum-urate"><a href="#TOC">Serum Urate</a></h2>
<pre><code>export POS_LABEL=serumUrateNatGen2013
export NEG_LABEL=1kg10000
export ANNOT_LABEL=annotation
export INDEX_LABEL=index3</code></pre>
<h2 id="body-mass-index"><a href="#TOC">Body-mass-index</a></h2>
<pre><code>export TAG_RSID_POS=$HOME/MEGA/2015_svmgwas/data/variant/literature_association/traits/bmiNatGen2013.rsid
export POS_LABEL=bmiNatGen2013
export NEG_LABEL=1kg100000
export ANNOT_LABEL=annotation
export INDEX_LABEL=index3</code></pre>
<h2 id="transmission-distortion"><a href="#TOC">Transmission distortion</a></h2>
<pre><code>export TAG_RSID_POS=${HOME}/data/2015_svmgwas/data/variant/GRASP/phenos108/transmissionDistortion.rsid
export POS_LABEL=transmissionDistortion
export NEG_LABEL=1kg100000
export ANNOT_LABEL=annotation
export INDEX_LABEL=index3</code></pre>
</body>
</html>
